### PR TITLE
Edit Entry Details in Management Interface

### LINKED ISSUE #34: Edit Entry Details in Management Interface
I want to be able to easily change my entry details (e.g., username and/or password).

## Acceptance Criteria
* Ability to change username
* Ability to change password
* Can see the password value as it's edited
* Asked to discard changes if any field has changed but not yet saved
   - when clicking "Cancel"
   - when clicking another entry
* Asked to confirm delete

## Visual Design
**InVision**
https://mozilla.invisionapp.com/share/ENOL84959AK#/338187075_Logins_-_Entry_Detail_-_Edit_-Light-
Long passwords: https://mozilla.invisionapp.com/share/ENOL84959AK#/340867863_Logins_-_Entry_Detail_-_Edit_-_Long_Password_-Light-

**Zeplin**
https://zpl.io/VkwEO1M
Long passwords: https://zpl.io/adrvOB7

### PR DESCRIPTION
**Pull request by lmorchard**
- Tweaks to editing fields to better match UX spec
- Item title now appears in header for editing
- Disabled origin field on editing
- Add delete button in header of edit view
- New delete.svg icon

Fixes #34

### CODE REVIEW

File: src/list/components/item-fields.js [hunk 1]
```diff
@@ -77,13 +77,20 @@ ItemFields.propTypes = {
...
       onChange: PropTypes.func.isRequired,
     };
   }
 
   componentDidMount() {
-    this._firstField.focus();
+    // Focus the first editable field...
+    if (this.props.fields.itemId) {
+      // For existing items, the first field is disabled.
+      this._secondField.focus();
```
Comment (linuxwolf) [inline src/list/components/item-fields.js]: I _think_ there's been some change in product requirements here (i.e., "Website address" is always editable).  cc @sandysage for confirmation.

If so, I would propose a follow-up issue; there be dragons if this field is editable.

File: src/list/manage/components/edit-item-details.js [hunk 2]
```diff
@@ -69,9 +70,26 @@ export default class EditItemDetails extends React.Component {
...
-        <Localized id={`item-details-heading-${newItem ? "new" : "edit"}`}>
-          <h1>eDIt iTEm</h1>
-        </Localized>
+        <header>
+          {newItem ? (
+            <Localized id={`item-details-heading-new`}>
+              <h1>nEw iTEm</h1>
+            </Localized>
+          ) : (
+            <React.Fragment>
```
Comment (linuxwolf) [inline src/list/manage/components/edit-item-details.js]: TIL about `React.Fragment` 🌈 🌟

File: src/list/manage/components/item-details.css [hunk 3]
```diff
@@ -7,15 +7,46 @@
   background-color: #f9f9fa;
 }
 
-.item-details > h1 {
+.item-details > header {
+  position: relative;
+}
+
+.item-details > h1,
```
Comment (linuxwolf) [inline src/list/manage/components/item-details.css]: will this conflict with identical work in #87 ?

File: src/list/components/item-fields.js [hunk 4]
```diff
@@ -96,18 +103,13 @@ export class EditItemFields extends React.Component {
...
-        </label>
         <label>
           <Localized id="item-fields-origin">
-            <LabelText>oRIGIn</LabelText>
+            <LabelText className={styles.firstLabel}>oRIGIn</LabelText>
           </Localized>
           <Localized id="item-fields-origin-input" attrs={{placeholder: true}}>
             <Input className={styles.input} type="text"
+                   disabled={!!fields.itemId}
                    placeholder="wWw.eXAMPLe.cOm"
```
Comment (linuxwolf) [inline src/list/components/item-fields.js]: The designs call for changing this to `"https://www.example.com" to match what we expect from the user.

**Comment by lmorchard** [comment]
> The biggest thing I've seem missing is a callout "bubble" when the "website address" field is focused, with instructions on what's expected

I think that's part of issue #30 as the "helper" in the specs

**Comment by linuxwolf** [comment]
> I think that's part of issue #30 as the "helper" in the specs. At least, that's where I started implementing it on a branch a few weeks ago

Ah, right.  Ignore this and carry on!

File: src/list/manage/components/edit-item-details.js (hunk 2 above)
Comment (lmorchard) [reply src/list/manage/components/edit-item-details.js]: Yeah, I think you can also use `<>` and `</>` but I like explicitly naming the `React.Fragment` component.

File: src/list/manage/components/item-details.css (hunk 3 above)
Comment (lmorchard) [reply src/list/manage/components/item-details.css]: Hmm, it probably will, but we should probably also just rebase / revise against whichever lands first? Not sure how to pre-emptively handle that well

**Comment by lmorchard** [comment]
Okay, made some tweaks to fix the hidden bit of the password field and updated the field placeholder. Not entirely sure how / if to be proactive about a conflict with #87

**Comment by lmorchard** [comment]
Took a look at #87 - [commented](https://github.com/mozilla-lockbox/lockbox-addon/pull/87#discussion_r260455222) to the effect that maybe we should land the other PR first and then I can revise this one to fix the conflicts

**Comment by linuxwolf** [comment]
Just checked it over, and it's working great!  At worst, the cursor disappears at the extreme edge, but arrowing left brings it back to visible.

File: src/list/components/item-fields.js (hunk 1 above)
Comment (linuxwolf) [reply src/list/components/item-fields.js]: discussed with @sandysage , and there is a change that had not been reflected into the desktop designs yet (it is in the mobile CRUD designs).

I'm going to file a follow up for this; editable hostname has complications/concerns we need to make sure we understand.

**Comment by lmorchard** [comment]
Rebased, resolved conflicts. Mainly little CSS tweaks, so I might merge after a successful test run. But, won't turn down another quick review

**Pull request merged**